python-modules/cis_publisher/cis_publisher/mozilliansorg_groups.py

```diff
@@ -0,0 +1,122 @@
+        new_image = get_nested(body, "dynamodb", "NewImage")
+        typ = get_nested(body, "eventName")
+        payload_user_id = get_nested(new_image, "user_id", "S")
+        if user_id != payload_user_id:
+            logger.error("missmatching user_ids {} {}".format(user_id, payload_user_id))
+            return None
+        groups = unpack_string_list(get_nested(new_image, "groups", "L", default=[]))
+        if user_id and typ and groups is not None:
+            return MozilliansorgGroupUpdate(typ, user_id, groups)
+        return None
+
+
+class MozilliansorgGroupsPublisher:
+    def __init__(self):
+        self.secret_manager = cis_publisher.secret.Manager()
+        self.publisher = cis_publisher.Publish([], login_method=None, publisher_name="mozilliansorg")
+
+    def publish(self, event):
+        """
+        Publish all resulting mozilliansorg group updates to CIS resulting from an event.
+        """
+        # Contruct MozillansGroupUpdates from all records of the event.
+        updates = [update for update in map(MozilliansorgGroupUpdate.from_record, event.get("Records", [])) if update]
+        # Create partial profile updates for all MozillansGroupUpdates and filter out no-ops.
+        update_profiles = [update_profile for update_profile in map(self._prepare_update, updates) if update_profile]
+        failed_updates = Queue()
+        for update_profile in update_profiles:
+            user_id = update_profile.user_id.value
+            qs = "/v2/user?user_id={}".format(quote_plus(user_id))
+            self.publisher._really_post_with_qs(user_id, qs, update_profile, failed_updates)
+        while not failed_updates.empty():
+            logger.warn("failed to update: {}".format(failed_updates.get()))
+            failed_updates.task_done()
+
+    def _prepare_update(self, update):
+        """
+        Construct a partial profile for a given update.
+        Returns a profile with user_id, active and the according updated and signed mozilliansorg access information or
+        None for a no-op.
+        """
+        current_profile = None
+        try:
+            current_profile = self.publisher.get_cis_user(update.user_id)
+        except PublisherError:
+            logger.info("No profile for {}, skipping!".format(update.user_id))
+            return None
+        if not hasattr(current_profile, "user_id"):
+            logger.info("No profile for {}, skipping!".format(update.user_id))
+            return None
+
+        logger.info("received profile for {}".format(current_profile.user_id.value))
+        update_profile = cis_profile.profile.User()
+        update_profile.user_id = current_profile.user_id
+        update_profile.active = current_profile.active
+        update_profile.access_information.mozilliansorg["values"] = {group: None for group in update.groups}
+        if (
+            update_profile.access_information.mozilliansorg["values"]
```

Comment:
@gdestuynder checking id anything changed now.

Reply:
im not sure what your comment means - but this looks alright. id also skip sending the same way.
you might want to call `sorted` for safety though as the array order is not guaranteed, i.e.

`if sorted(update_profile.access_information.mozilliansorg["values"]) == sorted(...)`